### PR TITLE
Various minor fixes and enhancements to S4J adapter

### PR DESCRIPTION
**Pull request by yabinmeng**
- Add message priority following a distribution pattern
- Fix the issue of message compression type not properly mapped 
- Change number connections to physical connections to the Pulsar server
- Pick up the latest S4J version

### CODE REVIEW

File: nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/S4JSpace.java [hunk 1]
```diff
@@ -217,45 +221,44 @@ public void incTxnBatchTrackingCnt() {
...
-            catch (Exception e) {
-                e.printStackTrace();
-            }
+        }
+        catch (JMSRuntimeException e) {
+            logger.error("Unable to initialize JMS connection factory with the following configuration parameters: {}", s4JClientConnInfo.toString());
+            throw new S4JAdapterUnexpectedException("Unable to initialize JMS connection factory with the following error message: " + e.getCause());
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```
Comment (jshook) [inline nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/S4JSpace.java:261]: Exceptions should not be swallowed here. If the intent is to ignore them during testing, then the error handler stack should be employed to do this as with `errors=MyErrorTypePattern.*:ignore,...`

File: nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/util/S4JClientConfConverter.java [hunk 2]
```diff
@@ -81,15 +81,14 @@ public static Map<String, Object> convertRawProducerConf(Map<String, String> pul
...
-                switch (StringUtils.upperCase(confVal)) {
-                    case "LZ4":
-                        compressionType = CompressionType.LZ4;
-                    case "ZLIB":
-                        compressionType = CompressionType.ZLIB;
-                    case "ZSTD":
-                        compressionType = CompressionType.ZSTD;
-                    case "SNAPPY":
-                        compressionType = CompressionType.SNAPPY;
+                if (StringUtils.equalsIgnoreCase(confVal, "LZ4")) {
```
Comment (jshook) [inline nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/util/S4JClientConfConverter.java:84]: This code could be much cleaner with an enum, _enum_.valueOf(...) and a newer switch form.

**Review by jshook — CHANGES_REQUESTED** [review]
Only one functional change requested, and an idea.

File: nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/S4JSpace.java (hunk 1 above)
Comment (yabinmeng) [reply nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/S4JSpace.java]: Thanks, this block is really not necessary and removed it.

File: nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/util/S4JClientConfConverter.java (hunk 2 above)
Comment (yabinmeng) [reply nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/util/S4JClientConfConverter.java]: Fixed as suggested

**Review by msmygit — APPROVED** [review]
lgtm 👍🏼

**Review by eolivelli — CHANGES_REQUESTED** [review]
In S4JClientConfConverter we miss to allow configuring these properties:
https://github.com/nosqlbench/nosqlbench/blob/b2e5f66d11074267e9ff1e0b98b783dd06e73841/nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/util/S4JClientConfConverter.java#L296-L311


Map.entry("jms.usePulsarAdmin","boolean"),
Map.entry("jms.enableJMSPriority","boolean"),

File: nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/util/S4JClientConfConverter.java [hunk 3]
```diff
@@ -75,27 +75,24 @@ public static Map<String, Object> convertRawProducerConf(Map<String, String> pul
...
-                } else if (StringUtils.equalsIgnoreCase(confVal, "SNAPPY")) {
-                    compressionType = CompressionType.SNAPPY;
-                }
-
-                s4jProducerConfObjMap.put(confKeyName, compressionType);
-            } else {
-                throw new S4JAdapterInvalidParamException(
-                    getInvalidConfValStr(confKeyName, confVal, "producer", expectedVal));
-            }
+        if (StringUtils.isNotBlank(confVal) && S4JAdapterUtil.isValidMsgCompressionTypeStr(confVal)) {
```
Comment (jshook) [inline nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/util/S4JClientConfConverter.java:79]: Looks like you could short-circuit this right to the eventual enum type.

**Review by jshook — APPROVED** [review]
Please update the enum type (remove the intermediate one if possible)
Otherwise, this can merge as is.

File: nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/S4JSpace.java [hunk 4]
```diff
@@ -100,7 +100,11 @@ public record JMSDestinationCacheKey(String contextIdentifier,
     private final ThreadLocal<Integer> txnBatchTrackingCnt = ThreadLocal.withInitial(() -> 0);
 
     // Represents the JMS connection
-    private PulsarConnectionFactory s4jConnFactory;
+//    private PulsarConnectionFactory s4jConnFactory;
```
Comment (bradfordcp) [inline nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/S4JSpace.java:103]: Nit - is there any reason why this is being left here vs removed?

**Pull request closed without merging**